test/unit/org/apache/cassandra/config/DurationSpecTest.java

```diff
@@ -59,102 +61,257 @@ public void testFromSymbol()
+        assertThatThrownBy(() -> new DurationSpec.IntMinutesBound("1491309d")).isInstanceOf(ConfigurationException.class)
+                                                                              .hasMessageContaining("Invalid duration: 1491309d. " +
+                                                                                                    "It shouldn't be more than 2147483646 in minutes");
+        assertThatThrownBy(() -> new DurationSpec.IntMinutesBound(2147483648L)).isInstanceOf(ConfigurationException.class)
+                                                                               .hasMessageContaining("Invalid duration: 2147483648 minutes. " +
+                                                                                                     "It shouldn't be more than 2147483646 in minutes");
+    }
+
     @Test
     public void testEquals()
     {
-        assertEquals(new DurationSpec("10s"), new DurationSpec("10s"));
-        assertEquals(new DurationSpec("10s"), new DurationSpec("10000ms"));
-        assertEquals(new DurationSpec("10000ms"), new DurationSpec("10s"));
-        assertEquals(DurationSpec.inMinutes(Long.MAX_VALUE), DurationSpec.inMinutes(Long.MAX_VALUE));
-        assertEquals(new DurationSpec("4h"), new DurationSpec("14400s"));
-        assertEquals(DurationSpec.inSecondsString("14400"), new DurationSpec("14400s"));
-        assertEquals(DurationSpec.inSecondsString("4h"), new DurationSpec("14400s"));
-        assertEquals(DurationSpec.inSecondsString("14400s"), new DurationSpec("14400s"));
-        assertEquals(DurationSpec.inHours(Long.MAX_VALUE),DurationSpec.inHours(Long.MAX_VALUE));
-        assertNotEquals(DurationSpec.inMinutes(Long.MAX_VALUE), DurationSpec.inMilliseconds(Long.MAX_VALUE));
-        assertNotEquals(new DurationSpec("0m"), new DurationSpec("10ms"));
+        assertEquals(new DurationSpec.LongNanosecondsBound ("10s"), new DurationSpec.LongNanosecondsBound ("10s"));
+        assertEquals(new DurationSpec.LongNanosecondsBound ("10s"), new DurationSpec.LongNanosecondsBound ("10000ms"));
+        assertEquals(new DurationSpec.LongNanosecondsBound ("10000ms"), new DurationSpec.LongNanosecondsBound ("10s"));
+        assertEquals(new DurationSpec.LongNanosecondsBound ("4h"), new DurationSpec.LongNanosecondsBound ("14400s"));
+        assertEquals(DurationSpec.LongNanosecondsBound .IntSecondsBound.inSecondsString("14400"), new DurationSpec.LongNanosecondsBound ("14400s"));
+        assertEquals(DurationSpec.LongNanosecondsBound .IntSecondsBound.inSecondsString("4h"), new DurationSpec.LongNanosecondsBound ("14400s"));
+        assertEquals(DurationSpec.LongNanosecondsBound .IntSecondsBound.inSecondsString("14400s"), new DurationSpec.LongNanosecondsBound ("14400s"));
+        assertNotEquals(new DurationSpec.LongNanosecondsBound ("0m"), new DurationSpec.LongNanosecondsBound ("10ms"));
```

Comment:
We can mark the method with `@SuppressWarnings("AssertEqualsBetweenInconvertibleTypes")` to get rid of the warnings.